src/Record/Record.Model/ProvenanceBuilder.cs

```diff
@@ -0,0 +1,89 @@
+using VDS.RDF;
+using VDS.RDF.Query.Algebra;
+using Graph = VDS.RDF.Query.Algebra.Graph;
+
+namespace Records;
+
+public record ProvenanceBuilder
+{
+    private Storage _storage = new();
+
+    public static Func<ProvenanceBuilder, ProvenanceBuilder> WithUsing(IEnumerable<string> used) => WithUsing(used.ToArray());
```

Comment:
I am not sure I understand why you have gone for `Func<ProvenanceBuilder, ProvenanceBuilder>` instead of `ProvenanceBuilder` as with the pattern in the `RecordBuilder`. 

Would you be able to explain it to me? :D